From 98fa54cc412790040fa5232fffae6c82e5c78022 Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex.wilson@joyent.com>
Date: Tue, 22 Dec 2015 17:42:12 -0800
Subject: [PATCH 36/36] Default to enabling "sandbox" when built
 --with-sandbox=solaris

The Solaris priv sandbox should be enabled by default, to preserve
old behaviour, and also because it does not have any significant
negative impacts on performance or otherwise.
---
 servconf.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/servconf.c b/servconf.c
index 245f2fd..5709978 100644
--- a/servconf.c
+++ b/servconf.c
@@ -472,7 +472,11 @@ fill_default_server_options(ServerOptions *options)
 
 	/* Turn privilege separation on by default */
 	if (use_privsep == -1)
+#ifdef USE_SOLARIS_PRIVS
+		use_privsep = PRIVSEP_ON;
+#else
 		use_privsep = PRIVSEP_NOSANDBOX;
+#endif
 
 #define CLEAR_ON_NONE(v) \
 	do { \
@@ -1108,7 +1112,11 @@ static const struct multistate multistate_gatewayports[] = {
 	{ NULL, -1 }
 };
 static const struct multistate multistate_privsep[] = {
+#ifdef USE_SOLARIS_PRIVS
+	{ "yes",			PRIVSEP_ON },
+#else
 	{ "yes",			PRIVSEP_NOSANDBOX },
+#endif
 	{ "sandbox",			PRIVSEP_ON },
 	{ "nosandbox",			PRIVSEP_NOSANDBOX },
 	{ "no",				PRIVSEP_OFF },
-- 
2.5.4 (Apple Git-61)

